std/algorithm/iteration.d

```diff
@@ -836,7 +838,7 @@ template each(alias pred = "a")
 
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (i, ref a; r)
+            foreach (ref i, ref a; r)
```

Comment:
Looks like it was already reported: https://issues.dlang.org/show_bug.cgi?id=14984.